ui/app/components/modals/new-account-modal.js

```diff
@@ -1,17 +1,88 @@
-const Component = require('react').Component
+const { Component } = require('react')
+const PropTypes = require('prop-types')
 const h = require('react-hyperscript')
-const inherits = require('util').inherits
-const connect = require('react-redux').connect
+const { connect } = require('react-redux')
 const actions = require('../../actions')
 
-function mapStateToProps (state) {
+class NewAccountModal extends Component {
+  constructor (props) {
+    super(props)
+    const { numberOfExistingAccounts = 0 } = props
+    const newAccountNumber = numberOfExistingAccounts + 1
```

Comment:
LGTM. For reference -- the keyring controller create nickname using the same logic [here](https://github.com/MetaMask/KeyringController/blob/master/index.js#L387)